src/contracts/examples/fastlane-online/OuterHelpers.sol

```diff
@@ -259,21 +276,23 @@ contract OuterHelpers is FastLaneOnlineInner {
         return sortedSolverOps;
     }
 
-    function _updateSolverReputation(
-        SolverOperation[] memory solverOps,
-        uint128 magnitude,
-        bool solversSuccessful
-    )
-        internal
-    {
+    function _updateSolverReputation(SolverOperation[] memory solverOps, uint128 magnitude) internal {
         uint256 _length = solverOps.length;
+        address _solverFrom;
+
         for (uint256 i; i < _length; i++) {
-            if (solversSuccessful) {
-                S_solverReputations[solverOps[i].from].successCost += magnitude;
+            _solverFrom = solverOps[i].from;
+
+            // winningSolver will be address(0) unless a winning solver fulfilled the swap intent.
+            if (_solverFrom == _getWinningSolver()) {
```

Comment:
Transient storage reads aren't as cheap as regular memory reads - can we preload into memory the winning solver above the for loop? (line 282 maybe?)

Reply:
Oof yes, good catch. Fixed here https://github.com/FastLane-Labs/atlas/pull/400/commits/80f32e99c2696b5819b2160cedba7c8b15159a8a